src/s3/types.rs

```diff
@@ -1781,17 +1769,17 @@ impl ReplicationConfig {
         let mut data = String::from("<ReplicationConfiguration>");
 
         if let Some(v) = &self.role {
-            data.push_str("<Status>");
-            data.push_str(&v);
-            data.push_str("</Status>");
+            data.push_str("<Role>");
```

Comment:
Firstly open an issue with complete information about current encode/decode issue of `ReplicationConfig`. I do not understand what you are trying to fix.